Userland/Utilities/mktemp.cpp

```diff
@@ -7,66 +7,60 @@
 #include <AK/LexicalPath.h>
 #include <AK/Random.h>
 #include <LibCore/ArgsParser.h>
-#include <errno.h>
+#include <LibCore/System.h>
+#include <LibMain/Main.h>
 #include <fcntl.h>
-#include <stdio.h>
-#include <string.h>
 #include <sys/stat.h>
 #include <unistd.h>
 
-constexpr const char* default_template = "tmp.XXXXXXXXXX";
+constexpr StringView default_template = "tmp.XXXXXXXXXX";
 
-static char* generate_random_filename(const char* pattern)
+static ErrorOr<String> generate_random_filename(String const& pattern)
 {
-    char* new_filename = strdup(pattern);
-    int pattern_length = strlen(pattern);
+    auto new_filename = StringBuilder { pattern.length() };
```

Comment:
:thinking: I think we prefer
```suggestion
    StringBuilder new_filename { pattern.length() };
```

(optional nitpick)